mlflow/utils/file_utils.py

```diff
@@ -229,6 +231,24 @@ def append_to(filename, data):
         handle.write(data)
 
 
+def make_tarfile(output_filename, source_dir, archive_name):
+    # Helper for filtering out modification timestamps
+    def _filter_timestamps(tar_info):
+        tar_info.mtime = 0
+        return tar_info
+    unzipped_filename = tempfile.mktemp()
+    try:
+        with tarfile.open(unzipped_filename, "w") as tar:
+            tar.add(source_dir, arcname=archive_name, filter=_filter_timestamps)
+        # When gzipping the tar, don't include the tar's filename or modification time in the
+        # zipped archive (see https://docs.python.org/3/library/gzip.html#gzip.GzipFile)
+        with gzip.GzipFile(filename="", fileobj=open(output_filename, 'wb'), mode='wb', mtime=0)\
```

Comment:
Btw, Python-based gzip is known to be very slow in comparison to the `GNU tar` utility. We may later want to go back to using that if we see this come up in practice.